oidc/config.go

```diff
@@ -0,0 +1,57 @@
+package oidc
+
+import "errors"
+
+type Config struct {
+	Enabled       bool     `koanf:"enabled"`
+	MetadataURL   string   `koanf:"metadata_url"`
+	ClientID      string   `koanf:"client_id"`
+	ClientSecret  string   `koanf:"client_secret"`
+	Scope         []string `koanf:"scope"`
+	SessionSecret string   `koanf:"session_secret"`
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Enabled: false,
+		Scope: []string{
+			"openid",
+			"profile",
+			"email",
+		},
+	}
+}
+
+func (c Config) Validate() error {
+	if !c.Enabled {
+		return nil
+	}
+
+	if c.MetadataURL == "" {
+		return errors.New("metadata_url is required")
+	}
+
+	if c.ClientID == "" {
+		return errors.New("client_id is required")
+	}
+
+	if c.ClientSecret == "" {
+		return errors.New("client_secret is required")
+	}
+
+	if c.SessionSecret == "" {
+		return errors.New("session_secret is required")
+	}
+
+	if c.Scope == nil {
```

Comment:
Should there always be at least 1 scope?

`len(c.Scope)` is nil-safe, so if you want to check whether there's elements in the slice, you can do

`if len(c.Scope) == 0`

`c.Scope == nil` check is semantically different, since slices can be nil, or not nil but with zero length.